src/chat/chat_line.rs

```diff
@@ -415,16 +415,24 @@ impl ChatLineRef {
         inner.label(id!(avatar_label)).set_text(text);
     }
 
-    pub fn set_message_text(&mut self, cx: &mut Cx, text: &str) {
+    pub fn set_message_text(&mut self, cx: &mut Cx, text: &str, is_streaming: bool) {
         let Some(mut inner) = self.borrow_mut() else {
             return;
         };
 
+        if is_streaming && !text.is_empty() {
+            let output = format!("{}{}", text, "●");
+            inner.text_input(id!(input)).set_text(&output.trim());
+            inner.label(id!(plain_text_message)).set_text(&output.trim());
+            inner.markdown(id!(markdown_message)).set_text(&output.trim());
+        } else {
+            inner.text_input(id!(input)).set_text(text.trim());
+            inner.label(id!(plain_text_message)).set_text(text.trim());
+            inner.markdown(id!(markdown_message)).set_text(text.trim());
+        }
```

Comment:
We're introducing an issue here. The user is not longer able to edit a message. It's happening because you moved the `inner.text_input(id!(input)).set_text(..)` out of the `ChatLineState::Editable | ChatLineState::NotEditable` arms of the `match inner.edition_state`. 

I feel that a better way to approach this change is to compute the change to the `text` (adding the ●) before calling this function. The caller has everything in place to do it.

Reply:
I moved the logic check back into the `match` , and it seems to have restored the previous effect. I tried adding the  ● in the `chat_panel`, but the code isn’t as clean with this approach.